app/models/article.rb

```diff
@@ -269,10 +269,8 @@ def remove_algolia_index
   end
 
   def delete_related_objects
-    index = Algolia::Index.new("searchables_#{Rails.env}")
-    index.delete_object("articles-#{id}")
-    index = Algolia::Index.new("ordered_articles_#{Rails.env}")
-    index.delete_object("articles-#{id}")
+    RemoveFromAlgoliaIndexJob.perform_later("searchables_#{Rails.env}", index_id)
```

Comment:
The objects used to be deleted from the index synchronously, but now they are deleted asynchronously. 
If you want to use the job here, you can use `perform_now` instead of `perform_later` to keep the existing logic.